src/main/java/com/khalsa_ji/ems/rest/Post.java

```diff
@@ -0,0 +1,137 @@
+     * @throws URISyntaxException URI Syntax Exception
+     * @see Employee
+     */
+
+    @PostMapping(value = "/employees", consumes = "application/json", produces = "application/json")
+    @ApiOperation(value = "Adds a new Employee", response = Employee.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Employee added successfully"),
+            @ApiResponse(code = 400, message = "Invalid details found for adding a new employee")
+    })
+    public ResponseEntity<Object> addEmployee(@RequestBody Employee employee) throws URISyntaxException {
+        //  Validating employee's name
+        switch(Validator.validateString(employee.getEmployeeName())) {
+            case nullString:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullString", EMPLOYEE_NAME, Locale.getDefault()));
+            case emptyString:       return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.emptyString", EMPLOYEE_NAME, Locale.getDefault()));
+            case invalidString:     return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.invalidString", EMPLOYEE_NAME, Locale.getDefault()));
+        }
+
+        //  Validating employee's job title(designation)
+        switch(Validator.validateString(employee.getJobTitle())) {
+            case nullString:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullString", JOB_TITLE, Locale.getDefault()));
+            case emptyString:       return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.emptyString", JOB_TITLE, Locale.getDefault()));
+            case invalidString:     return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.invalidString", JOB_TITLE, Locale.getDefault()));
+        }
+
+        //  Validating Manager ID
+        switch(Validator.validateID(employee.getManagerID())) {
+            case nullNumber:        return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.nullNumber", MANAGER, Locale.getDefault()));
+            case zero:              return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.code.zero", MANAGER, Locale.getDefault()));
+        }
+
+        Designation designation = designationService.getDesignation(employee.getJobTitle());
+
+        //  If designation with specified id does not exists.
+        if(designation.getLevelID() == -1)
+            return ResponseEntity.badRequest()
+                    .body(messageSource.getMessage("error.designation.notFound",
+                            new String[]{employee.getJobTitle()}, Locale.getDefault()));
+
+        employee.setJobID(designation);
+
+        Employee manager = employeeService.getEmployeeByID(employee.getManagerID());
```

Comment:
We should check if manager is not null, else I think it will throw NullPointerException on line 112.

Reply:
No, in any scenario the manager object in line 107 could not get initialised as null because in the implementation of the `public Employee getEmployeeByID(Long ID)` method defined in `com.khalsa_ji.ems.service.EmployeeService.java`, there is a validation implemented which checks that:
```
if  Employee findByEmployeeID(Long ID);
method in com.khalsa_ji.ems.repository.EmployeeRepository.java
returns null, in case no such employee is found possessing the given employee id,
then:
    return a default instance of Employee class which gets built by EmployeeBuilder class,
initialising the instance with default employee id of 0.
```
That's why in line 112, manager's employee id is checked for being 0 or not.
```
Manager has employee id as 0 means specified manager not found(invalid manager)
```